sigal/video.py

```diff
@@ -112,25 +118,78 @@ def generate_video(source, outname, settings, options=None):
         # biggest fitting dimension is width
         resize_opt = ['-vf', "scale=%i:trunc(ow/a/2)*2" % w_dst]
 
-    # do not resize if input dimensions are smaller than output dimensions
-    if w_src <= w_dst and h_src <= h_dst:
-        resize_opt = []
+    return resize_opt
+
+
+def _get_empty_if_none_else_variable(variable):
+    return [] if not variable else variable
 
+
+def generate_video_pass(converter, source, options, outname=None):
+    """Run a single pass of encoding.
+
+    :param source: source video
+    :param options: options to pass to encoder
+    :param outname: if multi-pass, this is None on the first pass
+    """
+    logger = logging.getLogger(__name__)
+    outname_opt = [] if not outname else [outname]
     # Encoding options improved, thanks to
     # http://ffmpeg.org/trac/ffmpeg/wiki/vpxEncodingGuide
     cmd = [converter, '-i', source, '-y']  # -y to overwrite output files
-    if options is not None:
-        cmd += options
-    cmd += resize_opt + [outname]
-
+    cmd += options + outname_opt
     logger.debug('Processing video: %s', ' '.join(cmd))
-    check_subprocess(cmd, source, outname)
+    check_subprocess(cmd, source, outname=outname)
+
+
+def generate_video(source, outname, settings):
+    """Video processor.
+
+    :param source: path to a video
+    :param outname: path to the generated video
+    :param settings: settings dict
+    :param options: array of options passed to ffmpeg
+
+    """
+    logger = logging.getLogger(__name__)
+
+    video_format = settings.get('video_format')
+    options = settings.get(video_format + '_options')
+    second_pass_options = settings.get(video_format + '_options_second_pass')
+    video_always_convert = settings.get('video_always_convert')
+    converter = settings['video_converter']
+
+    resize_opt = []
+    if settings.get("video_resize"):
+        src_dst_dimension_dict = get_dimensions(source, converter,
+                                                settings['video_size'])
+        resize_opt = get_resize_options(src_dst_dimension_dict)
+
+    base, src_ext = splitext(source)
+    base, dst_ext = splitext(outname)
+
+    if dst_ext == src_ext and not resize_opt and not video_always_convert:
+        logger.debug('For %s, the source and destination extension are the " \
+                        "same, there is no resizing to be done, and " \
+                        "video_always_convert is False, so the output is " \
+                        " being copied', outname)
+        shutil.copy(source, outname)
+        return
+
+    final_pass_options = _get_empty_if_none_else_variable(options) + resize_opt
+    if second_pass_options:
+        generate_video_pass(converter, source, final_pass_options)
+        final_second_pass_options = _get_empty_if_none_else_variable(
+                    second_pass_options) + resize_opt
+        generate_video_pass(converter, source,
+                            final_second_pass_options, outname)
```

Comment:
So when doing 2-pass encoding the idea is that ffmpeg will create a temp file in the first pass and find it when doing the second pass right ?

Reply:
That is correct, yes. I don't believe there is a way to change the filename from the default "ffmpeg2pass-0.log"